codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -190,6 +210,121 @@ public static void writeFrameHeader(ByteBuf out, int payloadLength, byte type,
     }
 
     /**
+     * Provides the ability to associate the outcome of multiple {@link ChannelPromise}
+     * objects into a single {@link ChannelPromise} object.
+     */
+    static class SimpleChannelPromiseAggregator extends DefaultChannelPromise {
+        private final ChannelPromise promise;
+        private int expectedCount;
+        private int successfullCount;
+        private int failureCount;
+        private boolean doneAllocating;
+
+        SimpleChannelPromiseAggregator(ChannelPromise promise, Channel c) {
+            super(c);
+            assert promise != null;
+            this.promise = promise;
+        }
+
+        SimpleChannelPromiseAggregator(ChannelPromise promise, Channel c, EventExecutor e) {
```

Comment:
Actually I am using this constructor, and not the other.  They are both present just to pass on to the super class, but I could get rid of the other one I think.